x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilter.java

```diff
@@ -297,41 +297,47 @@ private Map<String, List<FieldInferenceRequest>> createFieldInferenceRequests(Bu
                 final Map<String, Object> docMap = indexRequest.sourceAsMap();
                 boolean hasInput = false;
                 for (var entry : fieldInferenceMetadata.getFieldInferenceOptions().entrySet()) {
-                    String field = entry.getKey();
                     String inferenceId = entry.getValue().inferenceId();
-                    var value = XContentMapValues.extractValue(field, docMap);
-                    if (value == null) {
-                        continue;
-                    }
-                    if (inferenceResults.get(item.id()) == null) {
-                        inferenceResults.set(
-                            item.id(),
-                            new FieldInferenceResponseAccumulator(
+                    String fieldName = entry.getKey();
+                    for (var sourceField : entry.getValue().sourceFields()) {
```

Comment:
Now we iterate on all the source fields for retrieving inference - so we're adding an additional loop here